File: ueueo-framework/ueueo-authorization-abstractions/src/main/java/com/ueueo/authorization/PermissionsRequirement.cs
using JetBrains.Annotations;
using Microsoft.AspNetCore.Authorization;

namespace Volo.Abp.Authorization;

public class PermissionsRequirement : IAuthorizationRequirement
{
    public String[] PermissionNames;//  { get; }

    public boolean RequiresAll;//  { get; }

    public PermissionsRequirement(@NonNull String[] permissionNames, boolean requiresAll)
    {
        Objects.requireNonNull(permissionNames, nameof(permissionNames));

        PermissionNames = permissionNames;
        RequiresAll = requiresAll;
    }

    @Override public String toString()
    {
        return $"PermissionsRequirement: {string.Join(", ", PermissionNames)}";
    }
}

File: ueueo-framework/ueueo-authorization-abstractions/src/main/java/com/ueueo/authorization/PermissionsRequirement.java
package com.ueueo.authorization;

import com.ueueo.authorization.microsoft.IAuthorizationRequirement;
import lombok.Getter;
import org.apache.commons.lang3.StringUtils;
import org.springframework.lang.NonNull;

import java.util.List;
import java.util.Objects;

@Getter
public class PermissionsRequirement implements IAuthorizationRequirement {
    private List<String> permissionNames;

    private boolean requiresAll;

    public PermissionsRequirement(@NonNull List<String> permissionNames, boolean requiresAll) {
        Objects.requireNonNull(permissionNames);

        this.permissionNames = permissionNames;
        this.requiresAll = requiresAll;
    }

    @Override
    public String toString() {
        return String.format("PermissionsRequirement: %s", StringUtils.join(permissionNames, ", "));
    }
}
